discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -84,45 +102,43 @@ def get_host_assigned_roles(self):
     def get_operators(self):
         return self.api_client.get_cluster_operators(self.id)
 
-    def generate_image(self, ssh_key=env_variables['ssh_public_key']):
-        self.api_client.generate_image(cluster_id=self.id, ssh_key=ssh_key)
+    def generate_image(self):
+        self.api_client.generate_image(cluster_id=self.id, ssh_key=self._config.ssh_public_key)
 
     @JunitTestCase()
     def generate_and_download_image(
             self,
-            iso_download_path=env_variables['iso_download_path'],
-            ssh_key=env_variables['ssh_public_key'],
```

Comment:
Need to be able to overwrite ssh_key for negative tests

Reply:
Why not override config value?

Reply:
i can do that, but not sure its what need to be done
see this test verifying ssh key validation: 
https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/blob/master/api_tests/test_validations.py#L711
WDYT?

Reply:
Added option to override